jmx/src/main/java/io/airlift/jmx/JmxAgent9.java

```diff
@@ -44,26 +46,71 @@ public JmxAgent9(JmxConfig config)
             throws IOException
     {
         int registryPort;
-        if (config.getRmiRegistryPort() == null) {
-            registryPort = NetUtils.findUnusedPort();
+        Integer existingRegistryPort = Integer.getInteger(JMX_REGISTRY_PORT);
+        Integer configuredRegistryPort = config.getRmiRegistryPort();
+        if (existingRegistryPort != null) {
+            if (configuredRegistryPort != null && !existingRegistryPort.equals(configuredRegistryPort)) {
+                throw new RuntimeException(format(
+                        "System property '%s=%s' does match configured RMI registry port %s",
+                        JMX_REGISTRY_PORT, existingRegistryPort, configuredRegistryPort));
+            }
+            if (existingRegistryPort == 0) {
```

Comment:
This compares an Integer with 0 potentially using identity equality, depending on whether 0 boxes to Integer or existingRegistryPort unboxes to its value.